lib/internal/Magento/Framework/App/Action/Action.php

```diff
@@ -3,28 +3,38 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+
 namespace Magento\Framework\App\Action;
 
+use Magento\Framework\App\ActionFlag;
 use Magento\Framework\App\RequestInterface;
+use Magento\Framework\App\Response\RedirectInterface;
 use Magento\Framework\App\ResponseInterface;
-use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\App\ViewInterface;
+use Magento\Framework\Event\ManagerInterface as EventManagerInterface;
 use Magento\Framework\Exception\NotFoundException;
+use Magento\Framework\Message\ManagerInterface as MessageManagerInterface;
+use Magento\Framework\ObjectManagerInterface;
+use Magento\Framework\Profiler;
+use Magento\Framework\UrlInterface;
 
 /**
  * Extend from this class to create actions controllers in frontend area of your application.
  * It contains standard action behavior (event dispatching, flag checks)
  * Action classes that do not extend from this class will lose this behavior and might not function correctly
  *
- * TODO: Remove this class. Allow implementation of Action Controllers by just implementing Action Interface.
+ * @TODO: Remove this class. Allow implementation of Action Controllers by just implementing Action Interface.
```

Comment:
This message is a little confusing. Thanks to this PR we will `Allow implementation of Action Controllers by just implementing Action Interface.`. Let's just remove it. `@deprecated`  should be good enough.

Reply:
That is fine. I am going to remove `@TODO`, also I would like to change wording for `@deprecated`
```php
* @deprecated 2.4.0, use \Magento\Framework\App\ActionInterface
```